instrument_api/src/test/java/com/example/instrument_api/InstrumentControllerIntegrationTests.java

```diff
@@ -1,4 +1,199 @@
+    Instrument instrument2 = new Instrument(0,"Flute","The flute is the oldest of all instruments that produce pitched sounds (not just rhythms), and was originally made from wood, stone, clay or hollow reeds like bamboo. Modern flutes are made of silver, gold or platinum; there are generally 2 to 4 flutes in an orchestra. A standard flute is a little over 2 feet long and is often featured playing the melody. You play the flute by holding it sideways with both hands and blowing across a hole in the mouthpiece, much like blowing across the top of a bottle. Your fingers open and close the keys, which changes the pitch.", "1981","Woodwind", Collections.singleton(mainList.get(2)));
+    Instrument instrument3 = new Instrument(0,"Trumpet","If you think the brass family got its name because the instruments are made of brass, you're right! This family of instruments can play louder than any other in the orchestra and can also be heard from far away. Although their early ancestors are known to have been made of wood, tusks, animal horns or shells, today's modern instruments are made entirely of brass. Brass instruments are essentially very long pipes that widen at their ends into a bell-like shape. The pipes have been curved and twisted into different shapes to make them easier to hold and play.", "1982","Brass", Collections.singleton(mainList.get(1)));
+    Instrument instrument4 = new Instrument(0,"Piano","This class of musical instruments requires you to blow into a specific wind instrument by following an order to ensure that the sound that you desire is produced. The instruments can be expected to work depending on the principles of frequencies, sound waves, acoustics, resonance and harmonics. The pitch of the produced sound when you start blowing the instrument is actually dependent on the length of the air column through which the waves of the sounds vibrate", "1980","Percussion", Collections.singleton(mainList.get(0)));
+
+
+    @BeforeEach
+    public void beforeAllTests()
+    {
+        mockMvc = MockMvcBuilders.webAppContextSetup(webApplicationContext).build();
+        if(instrumentRepository.count()==0)
```

Comment:
Dit gaat onverwachte effecten geven als er al data in de database staat: de gegevens van de vorige keer, ook als die aangepast zijn, gaan erin staan en deze worden niet toegevoegd, dus het kan dat de tests daardoor niet meer gaan werken.